python/plugins/processing/algs/qgis/Climb.py

```diff
@@ -0,0 +1,255 @@
+    MINELEVATTRIBUTE = 'minelev'
+    MAXELEVATTRIBUTE = 'maxelev'
+
+    def name(self):
+        return 'climbalongline'
+
+    def displayName(self):
+        return self.tr('Climb Along Line')
+
+    def group(self):
+        return self.tr('Vector analysis')
+
+    def groupId(self):
+        return 'vectoranalysis'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(
+            QgsProcessingParameterFeatureSource(
+                self.INPUT,
+                self.tr('Input (line) layer'),
+                [QgsProcessing.TypeVectorLine]
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterFeatureSink(
+                self.OUTPUT,
+                self.tr('Climb layer')
+            )
+        )
+
+        self.addOutput(
+            QgsProcessingOutputNumber(
+                self.TOTALCLIMB,
+                self.tr('Total climb')
+            )
+        )
+
+        self.addOutput(
+            QgsProcessingOutputNumber(
+                self.TOTALDESCENT,
+                self.tr('Total descent')
+            )
+        )
+
+        self.addOutput(
+            QgsProcessingOutputNumber(
+                self.MINELEVATION,
+                self.tr('Minimum elevation')
+            )
+        )
+
+        self.addOutput(
+            QgsProcessingOutputNumber(
+                self.MAXELEVATION,
+                self.tr('Maximum elevation')
+            )
+        )
+
+    def processAlgorithm(self, parameters, context, feedback):
+
+        source = self.parameterAsSource(
+            parameters,
+            self.INPUT,
+            context
+        )
+
+        fcount = source.featureCount()
+
+        hasZ = QgsWkbTypes.hasZ(source.wkbType())
+
+        if not hasZ:
+            feedback.reportError(self.tr('The layer does not have Z values. If you have a DEM, use the Drape algorithm to extract Z values.'))
+            return
+
+        thefields = QgsFields()
+        climbindex = -1
+        descentindex = -1
+        fieldnumber = 0
+
+        # Skip fields with names that are equal to the generated ones
+        for field in source.fields():
+            if field.name() == self.CLIMBATTRIBUTE:
+                feedback.pushWarning(self.tr(
+                                '{attr_name} attribute found and removed'.format(
+                                    attr_name=self.CLIMBATTRIBUTE)
+                                    )
+                                )
+                climbindex = fieldnumber
+            elif field.name() == self.DESCENTATTRIBUTE:
+                feedback.pushWarning(self.tr(
+                                '{attr_name} attribute found and removed'.format(
+                                    attr_name=self.DESCENTATTRIBUTE)
+                                    )
+                                )
+                descentindex = fieldnumber
+            else:
```

Comment:
Sorry, but I have forgotten to include elifs for self.MINELEVATTRIBUTE and self.MAXELEVATTRIBUTE:

            elif field.name() == self.MINELEVATTRIBUTE:
                feedback.pushWarning(self.tr(
                                '{attr_name} attribute found and removed'.format(
                                    attr_name=self.MINELEVATTRIBUTE)
                                    )
                                )
                minelevindex = fieldnumber
            elif field.name() == self.MAXELEVATTRIBUTE:
                feedback.pushWarning(self.tr(
                                '{attr_name} attribute found and removed'.format(
                                    attr_name=self.MAXELEVATTRIBUTE)
                                    )
                                )
                maxelevindex = fieldnumber